docs/guide/using_existing_contracts.rst

```diff
@@ -37,47 +37,47 @@ If you do not have ABI statically, but you know the interface of the contract on
 Fees
 ----
 
-starknet.py requires you to specify amount of Wei you
-are willing to pay either when making ``.invoke()`` transactions or when preparing
-function calls with ``.prepare()``.
+Starknet.py requires you to specify amount of Wei (for V1 transaction) or Fri (for V3 transaction) you
+are willing to pay when executing either ``.invoke_v1()`` or ``.invoke_v3()`` transactions. It's also possible to optionally define the transaction fee
+when preparing function calls with ``.prepare_invoke_v1()`` or ``.prepare_invoke_v3()``.
 
 .. code-block:: python
 
-    await contract.functions["put"].invoke(k, v, max_fee=5000)
+    await contract.functions["put"].invoke_v1(k, v, max_fee=5000)
 
-When max_fee is specified when preparing a call, you can invoke it without
-``max_fee``.
+When max_fee is specified when preparing a call, you can invoke it without ``max_fee``. It works similarly for ``l1_resource_bounds`` and ``prepare_invoke_v3()``.
```

Comment:
I don't quite understand it to be honest. what does `when max_fee is specified (...) you can invoke it without max_fee` mean?

Reply:
When `max_fee` is specified in either `prepare_invoke_v1` or `prepare_invoke_v3`, the prepared call can subsequently be invoked without redefining `max_fee`, simply by using `prepared_call.invoke()`, rather than `prepared_call.invoke(max_fee=...)`. I agree the wording isn't optimal, I will try to rephrase it.